src/main/java/com/onegini/examples/resourcegateway/web/ResourcesController.java

```diff
@@ -43,27 +46,57 @@
   @RequestMapping(value = "/devices", method = RequestMethod.GET)
   public ResponseEntity<?> getDevices(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
 
+    final TokenValidationResult tokenValidationResult = validateAccessToken(authorizationHeader);
+
+    final boolean readScopeNotGranted = !scopeValidationService.isReadScopeGranted(tokenValidationResult.getScopes());
+    if (readScopeNotGranted) {
+      return buildInvalidScopeResponse();
+    }
+
+    return deviceApiRequestService.getDevices(tokenValidationResult.getUserId());
+  }
+
+  @RequestMapping(value = "/application-details", method = RequestMethod.GET)
+  public ResponseEntity<?> getApplicationDetails(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
+
+    final TokenValidationResult tokenValidationResult = validateAccessToken(authorizationHeader);
+
+    final boolean applicationDetailsScopeNotGranted = !scopeValidationService.isApplicationDetailsScopeGranted(tokenValidationResult.getScopes());
+    if (applicationDetailsScopeNotGranted) {
+      return buildInvalidScopeResponse();
+    }
+
+    final ApplicationDetails applicationDetails = new ApplicationDetails(tokenValidationResult);
+    return new ResponseEntity<Object>(applicationDetails, OK);
+  }
+
+  private TokenValidationResult validateAccessToken(final String authorizationHeader) {
     final Optional<String> accessToken = accessTokenExtractor.extractFromHeader(authorizationHeader);
     final boolean noAccessTokenFound = !accessToken.isPresent();
     if (noAccessTokenFound) {
-      LOGGER.warn("Provided authorization header not properly formatted.");
-      return buildBadRequestResponse();
+      throw new BadRequestException("Provided authorization header not properly formatted.");
     }
 
     final Optional<TokenValidationResult> optionalTokenValidationResult = tokenValidationService.validateAccessToken(accessToken.get());
     final boolean invalidAccessToken = !optionalTokenValidationResult.isPresent();
```

Comment:
the following code can be replaced by: return optionalTokenValidationResult.orElseThrow(() -> new InvalidAccessTokenException()); , please replace it for other execptions ;)